neo/rawio/phyrawio.py

```diff
@@ -0,0 +1,259 @@
+    * Create a file in neo/io/ that ends with "io.py"
+    * Create a class that inherits both your RawIO class and BaseFromRaw class
+    * copy/paste from neo/io/exampleio.py
+
+  4.Step 4 : IO test
+    * create a file in neo/test/iotest with the same previous name with "test_" prefix
+    * copy/paste from neo/test/iotest/test_exampleio.py
+
+
+
+"""
+
+from .baserawio import (BaseRawIO, _signal_channel_dtype, _unit_channel_dtype,
+                        _event_channel_dtype)
+
+import numpy as np
+from pathlib import Path
+import re
+import csv
+import ast
+
+
+class PhyRawIO(BaseRawIO):
+    """
+    Class for reading Phy data.
+
+    Usage:
+        >>> import neo.rawio
+        >>> r = neo.rawio.PhyRawIO(dirname='/dir/to/phy/folder')
+        >>> r.parse_header()
+        >>> print(r)
+        >>> spike_timestamp = r.get_spike_timestamps(block_index=0,
+        ... seg_index=0, unit_index=0, t_start=None, t_stop=None)
+        >>> spike_times = r.rescale_spike_timestamp(spike_timestamp, 'float64')
+
+    """
+    extensions = []
+    rawmode = 'one-dir'
+
+    def __init__(self, dirname=''):
+        BaseRawIO.__init__(self)
+        self.dirname = dirname
+
+    def _source_name(self):
+        return self.dirname
+
+    def _parse_header(self):
+        phy_folder = Path(self.dirname)
+
+        self._spike_times = np.load(phy_folder / 'spike_times.npy')
+        self._spike_templates = np.load(phy_folder / 'spike_templates.npy')
+
+        if (phy_folder / 'spike_clusters.npy').is_file():
+            self._spike_clusters = np.load(phy_folder / 'spike_clusters.npy')
+        else:
+            self._spike_clusters = self._spike_templates
+
+        # TODO: Add this when array_annotations are ready
+        # if (phy_folder / 'amplitudes.npy').is_file():
+        #     amplitudes = np.squeeze(np.load(phy_folder / 'amplitudes.npy'))
+        # else:
+        #     amplitudes = np.ones(len(spike_times))
+        #
+        # if (phy_folder / 'pc_features.npy').is_file():
+        #     pc_features = np.squeeze(np.load(phy_folder / 'pc_features.npy'))
+        # else:
+        #     pc_features = None
+
+        # SEE: https://stackoverflow.com/questions/4388626/
+        #  python-safe-eval-string-to-bool-int-float-none-string
+        if (phy_folder / 'params.py').is_file():
+            with (phy_folder / 'params.py').open('r') as f:
+                contents = f.read()
+            metadata = dict()
+            contents = contents.replace('\n', ' ')
+            pattern = re.compile(r'(\S*)[\s]?=[\s]?(\S*)')
+            elements = pattern.findall(contents)
+            for key, value in elements:
+                metadata[key.lower()] = ast.literal_eval(value)
+
+        self._sampling_frequency = metadata['sample_rate']
+
+        clust_ids = np.unique(self._spike_clusters)
+        self.unit_labels = list(clust_ids)
+
+        self._t_start = 0.
+        self._t_stop = max(self._spike_times) / self._sampling_frequency
```

Comment:
I guess there is no other solution for this.
It is ideal. Maybe you should add it in the class doc something like : `"t_stop" is not the real stop because the format don't handle it.`

Reply:
This is outdated. I did `max(self._spike_times).item() / self._sampling_frequency` in a later commit